keps/sig-api-machinery/5073-declarative-validation-with-validation-gen/README.md

```diff
@@ -0,0 +1,1308 @@
+    *   GA: Since validation is used to persist objects to storage, if there was a mistake and it permits objects that should not be persisted, rollback may be impacted by preventing updates on that object until the error is fixed.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+*   DeclarativeValidation: If the port to declarative validation is not accurate, during an update, old parts of an object may be seen as invalid and need to be ratcheted.
+
+###### Are there any tests for feature enablement/disablement?
+
+For targeting beta, this enhancement will have tests for:
+
+*   Yes, this enhancement specifically tests that all of the current validation tests have no differences when with `DeclarativeValidation` enabled or disabled.  In this way we can be confident that when `DeclarativeValidation` is enabled there is no functional difference in how objects are validated.
+
+### Rollout, Upgrade and Rollback Planning
+
+        This section must be completed when targeting beta to a release.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Beta & GA:
+
+A rollout can fail by being too strict with updates. If the ported declarative validations are stricter than native validations then workloads may not be able to execute their update operations.
+
+A rollback can fail if our declarative validations are too loose. Workloads wont be able to update objects with an invalid field until the object is corrected.
+
+###### What specific metrics should inform a rollback?
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+        This section must be completed when targeting beta to a release.
+
+        For GA, this section is required: approvers should be able to confirm the previous answers based on experience in the field.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+```bash
+    $ kubectl get --raw /metrics | grep DeclarativeValidation
+    # look for if the featuregate is enabled from /metrics
+kubernetes_feature_enabled{name="DeclarativeValidation",stage=""} 1
+```
+
+###### How can someone using this feature know that it is working for their instance?
+
+1. Events
+    *   Event Reason:
+2. API .status
+    *   Condition name:
+    *   Other field:
+3. Other (treat as last resort)
+    *   Details:
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+4. Metrics
+    *   Metric name: apiserver_request_duration_seconds
```

Comment:
also needs to list the `DeclarativeValidationMismatchMetrics` metric so we'll remember to flag any instances for feedback before promotion.

Reply:
Done, added the `DeclarativeValidationMismatchMetrics` metric to the list as well:

    *   Metric name: `declarative_validation_mismatch_metric`